network/manager.go

```diff
@@ -395,9 +395,19 @@ func (nm *networkManager) CreateEndpoint(cli apipaClient, networkID string, epIn
 	if err != nil {
 		return err
 	}
+	// any error after this point should also clean up the endpoint we created above
+	defer func() {
+		if err != nil {
+			delErr := nw.deleteEndpoint(nm.netlink, nm.plClient, nm.netio, nm.nsClient, nm.iptablesClient, ep.Id)
```

Comment:
can we log error before calling delete.. otherwise it will be confusing when reading logs